Assets/Scripts/Wallet/IWallet.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace Hermitage
+{
+    namespace Economy
+    {
+        using System;
+        using System.Collections.Generic;
+        
+        public interface IWallet
+        {
+            event EventHandler OnCoinAdded;
+            event EventHandler OnCoinRemoved;
+            bool IsFull { get; }
+            int Value { get; }
+            int RemainingCapacity { get; }
+            bool CanAfford(int cost);
+            ICoin[] Charge(int cost);
+            void Pay(params ICoin[] payments);
```

Comment:
These method names don't immediately imply a clear purpose/context. My brain is expecting words that reflect the fact that these two methods are opposites to each other. Using (admittedly boring) words like "Dispense" and "Collect" or "Remove" and "Add" would be more viscerally indicative of the actions performed. Before I saw the method code, I thought Pay would be giving payment or removing coins.